src/app/common/subscriptions/SubscriptionDetails.vue

```diff
@@ -101,32 +73,38 @@ const props = defineProps({
   },
 })
 
-const detailsIterator = computed<Record<string, Record<string, any>>>(() => {
-  let details
-  if (props.isDiscoverySubscription) {
-    const { lastUpdateTime, total, ...restDetails } = props.details.status
+type StatusRow = {
+  type: string
+  responsesSent: number
+  responsesAcknowledged: number
+  responsesRejected: number
+}
 
-    details = restDetails
-  }
+const statuses = computed<StatusRow[]>(() => {
+  let status: Record<string, DiscoveryServiceStats>
 
-  if (props.details.status?.stat) {
-    details = props.details.status?.stat
+  if ('controlPlaneInstanceId' in props.subscription) {
+    const { lastUpdateTime, total, ...discoverySubscriptionStatus } = props.subscription.status
+    status = discoverySubscriptionStatus
+  } else {
+    status = props.subscription.status?.stat ?? {}
   }
 
-  return details
-})
-
-function formatValue(value: string): string {
-  return value ? parseInt(value, 10).toLocaleString('en').toString() : '0'
-}
-
-function formatError(value: string): string {
-  if (value === '--') {
-    return 'error calculating'
+  if (!status) {
+    return []
   }
 
-  return value
-}
+  return Object.entries(status).map(([type, stat]) => {
+    const { responsesSent = '0', responsesAcknowledged = '0', responsesRejected = '0' } = stat
+
+    return {
+      type,
+      responsesSent: parseInt(responsesSent),
+      responsesAcknowledged: parseInt(responsesAcknowledged),
+      responsesRejected: parseInt(responsesRejected),
+    }
+  })
+})
```

Comment:
Will be lovely once we have our data layer in here so we don't have to do this all the time. I was wondering why these need to be integers? I can't see any math in here, looks like we are just printing them to the page as strings?

Reply:
Ah, that’s fair enough. I changed this to keep strings here (though not sure why they’re strings in the API to begin with).